src/customer/strategies/amazon-maxo/amazon-maxo-customer-strategy.spec.ts

```diff
@@ -0,0 +1,216 @@
+
+                await strategy.initialize(customerInitializeOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('fails to initialize the strategy if no AmazonMaxoCustomerInitializeOptions is provided ', async () => {
+                customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions(Mode.Incomplete);
+
+                await expect( () => strategy.initialize(customerInitializeOptions) ).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no methodid is supplied', async () => {
+                customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions(Mode.UndefinedMethodId);
+
+                await expect( () => strategy.initialize(customerInitializeOptions) ).toThrow(MissingDataError);
+            });
+
+            it('fails to create button if method is not assigned', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(undefined);
+                customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions();
+
+                await expect(strategy.initialize(customerInitializeOptions) ).rejects.toThrow(MissingDataError);
+            });
+
+            it('fails to create button if config is not initialized', async () => {
+                jest.spyOn(store.getState().config, 'getStoreConfig').mockReturnValue(undefined);
+                customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions();
+
+                await expect(strategy.initialize(customerInitializeOptions) ).rejects.toThrow(MissingDataError);
+            });
+
+            it('fails to create button if merchantId is not supplied', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(getPaymentMethodMockUndefinedMerchant());
+                customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions();
+
+                await expect(strategy.initialize(customerInitializeOptions) ).rejects.toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', async () => {
+                customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions(Mode.InvalidContainer);
+
+                await expect(strategy.initialize(customerInitializeOptions) ).rejects.toThrow(InvalidArgumentError);
+
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let customerInitializeOptions: CustomerInitializeOptions;
+
+        beforeEach(() => {
+            customerInitializeOptions = getAmazonMaxoCustomerInitializeOptions();
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
+            await strategy.initialize(customerInitializeOptions);
+
+            strategy.deinitialize();
```

Comment:
maybe add a test of what's expected to happen when you call `deinitialize ` without calling `initialize `

Reply:
Test was implemented, with the same logic expecting not to be Null.